### PR TITLE
Remove external funcy dependencies

### PR DESCRIPTION
**Pull request by Mogost**
It is necessary to keep as little as possible external dependencies.
`funcy `doesn't make the code much easier.

### CODE REVIEW

File: abnorm/adapters/django1p6.py [hunk 1]
```diff
@@ -118,11 +116,11 @@ def get_field_backwards_name(self, field):
         elif isinstance(rf, ManyToManyField):
             return rf.related.get_accessor_name()
         elif isinstance(rf, GenericRelation):
-            return first([
+            return next(iter([
                 f for f in self.get_model_fields(
                     descriptor.field.rel.to)
                 if self._is_matching_generic_foreign_key(descriptor.field, f)
-            ]).name
+            ]), None).name
```
Comment (tsouvarev) [inline abnorm/adapters/django1p6.py]: maybe add `first` as helper in utils.py for better readablility?

Comment (Mogost) [reply abnorm/adapters/django1p6.py]: In general, my goal in this pull request was to get rid of the extra dependency.
I agree to the move the code in the utils, it is reasonable.
But I wrote just the equivalent code. `first()` returns first element or None. If there is no element `AttributeError` will be raised upon access name.
We can rewrite this code to something like:
```
          return [
                f for f in self.get_model_fields(descriptor.field.rel.to)
                if self._is_matching_generic_foreign_key(descriptor.field, f)
            ][0].name
```
then `IndexError` will be raised and I see no difference in this case. But just indexing looks simpler.

Comment (tsouvarev) [reply abnorm/adapters/django1p6.py]: It really is equivalent code

It's just you are changing a lot of code across the project where you could change only one line of import

Comment (Mogost) [reply abnorm/adapters/django1p6.py]: Please recheck with new commits.
I really don't want to call `first()` without need.
That will take time without benefits.
```
%timeit [1,2,3][0]
246 ns ± 10.2 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

%timeit next(iter([1,2,3]), None)
614 ns ± 13.3 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Comment (tsouvarev) [reply abnorm/adapters/django1p6.py]: looks good, thank you

**Pull request closed without merging**